cfgmgr/vlanmgr.cpp

```diff
@@ -675,6 +678,167 @@ void VlanMgr::doVlanMemberTask(Consumer &consumer)
     }
 }
 
+void VlanMgr::doVlanStackTask(Consumer& consumer)
+{
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        auto& t = it->second;
+
+        string key = kfvKey(t);
+
+        vector<string> keys = tokenize(key, config_db_key_delimiter);
+
+        if (keys.size() != 2)
+        {
+            SWSS_LOG_ERROR("Unknown VLAN Stacking config %s.", key.c_str());
+            it = consumer.m_toSync.erase(it);
+            continue;
+        }
+
+        if (!isValidVlanId(keys.at(1), 1, 4094))
+        {
+            SWSS_LOG_ERROR("Unknown VLAN Stacking config %s.", key.c_str());
+            it = consumer.m_toSync.erase(it);
+            continue;
+        }
+
+        string new_key = keys.at(0) + DEFAULT_KEY_SEPARATOR + keys.at(1);
+
+        string op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            uint8_t s_vlan_priority = 0;
+            vector<uint16_t> c_vlanids;
+
+            for (auto itp : kfvFieldsValues(t))
+            {
+                string attr_name = fvField(itp);
+                string attr_value = to_upper(fvValue(itp));
+
+                if (attr_name == "c_vlanids")
+                {
+                    auto rv = parseVlanList(attr_value, c_vlanids);
+                    if (rv == false)
+                    {
+                        goto vlan_stacking_parse_failed;
+                    }
+                }
+                else if (attr_name == "s_vlan_priority")
+                {
+                    try
+                    {
+                        s_vlan_priority = (uint8_t)stoi(attr_value);
+                    }
+                    catch (std::invalid_argument& e)
+                    {
+                        goto vlan_stacking_parse_failed;
+                    }
+
+                    if (s_vlan_priority < 0 || s_vlan_priority > 7)
```

Comment:
s_vlan_priority is unsigned int. It cant be "< 0". Need to relook at this check?

Reply:
BTW 0xFF is a valid value right?

Reply:
no, 0xFF is not valid value